boa/src/builtins/regexp/mod.rs

```diff
@@ -283,38 +266,33 @@ impl RegExp {
 
         // 5. If F contains any code unit other than "g", "i", "m", "s", "u", or "y"
         //    or if it contains the same code unit more than once, throw a SyntaxError exception.
-        let mut global = false;
-        let mut ignore_case = false;
-        let mut multiline = false;
-        let mut dot_all = false;
-        let mut unicode = false;
-        let mut sticky = false;
+        let mut flags = RegExpFlags::default();
         for c in f.chars() {
```

Comment:
I would implement `FromStr` for `RegexFlags`, and use the version of `parse_regex_flags()` in the lexer, which iterates through bytes instead of characters, and should therefore be more efficient. Would also reduce duplicate code.